### PR TITLE
Fix links in the case when there are no incoming query params

### PR DESCRIPTION
**Pull request by imtayadeway**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1514097

IMO the surrounding code could benefit from some more sophisticated URI parsing, but I wanted to focus on fixing the bug first and leave that as a follow up.

@miq-bot add-label bug, gaprindashvili/yes
@miq-bot assign @abellotti 

/cc @jntullo

### CODE REVIEW

File: lib/api/links_builder.rb [hunk 1]
```diff
@@ -42,7 +42,10 @@ def format_href(new_offset)
       if href.include?("offset")
         href.sub("offset=#{offset}", "offset=#{new_offset}")
       else
-        href + "&offset=#{new_offset}"
+        result = URI.parse(href)
+        query_params = result.query.to_s.split("&") + ["offset=#{new_offset}"]
+        result.query = query_params.join("&")
+        result.to_s
```
Comment (abellotti) [inline lib/api/links_builder.rb]: you'd think URI would have a method to append something to a query.

maybe something like this:

```
result = URI.parse(href)
result.query = [result.query, "offset=#{new_offset}"].compact.join('&')
result.to_s
```

Thanks.

**Comment by imtayadeway** [comment]
@abellotti done. Merge when 🍏 ?

**Comment by abellotti** [comment]
Will do. Thanks @imtayadeway for fixing this !! 🎵

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 8f7a4a6c4f25e00ec2a9ef40013efc0b8df5fbf8
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Tue Nov 21 12:32:48 2017 -0500

    Merge pull request #221 from imtayadeway/bug/fix-links
    
    Fix links in the case when there are no incoming query params
    (cherry picked from commit 1e91ed2b7ba2d20726ac8d76cdeb69e7bc8f7f3d)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1517835
```

**Pull request merged**